contracts/traderjoe/FarmLensV2.sol

```diff
@@ -0,0 +1,364 @@
+        joeFactory = _joeFactory;
+        chefv2 = _chefv2;
+        chefv3 = _chefv3;
+        bmcj = _bmcj;
+    }
+
+    /// @notice Returns price of avax in usd.
+    function getAvaxPrice() public view returns (uint256) {
+        uint256 priceFromWavaxUsdte = _getAvaxPrice(IJoePair(wavaxUsdte)); // 18
+        uint256 priceFromWavaxUsdce = _getAvaxPrice(IJoePair(wavaxUsdce)); // 18
+        uint256 priceFromWavaxUsdc = _getAvaxPrice(IJoePair(wavaxUsdc)); // 18
+
+        uint256 sumPrice = priceFromWavaxUsdte.add(priceFromWavaxUsdce).add(
+            priceFromWavaxUsdc
+        ); // 18
+        uint256 avaxPrice = sumPrice / 3; // 18
+        return avaxPrice; // 18
+    }
+
+    /// @notice Returns value of wavax in units of stablecoins per wavax.
+    /// @param pair A wavax-stablecoin pair.
+    function _getAvaxPrice(IJoePair pair) private view returns (uint256) {
+        (uint256 reserve0, uint256 reserve1, ) = pair.getReserves();
+
+        if (pair.token0() == wavax) {
+            reserve1 = reserve1.mul(_tokenDecimalsMultiplier(pair.token1())); // 18
+            return (reserve1.mul(1e18)) / reserve0; // 18
+        } else {
+            reserve0 = reserve0.mul(_tokenDecimalsMultiplier(pair.token0())); // 18
+            return (reserve0.mul(1e18)) / reserve1; // 18
+        }
+    }
+
+    /// @notice Get the price of a token in Usd.
+    /// @param tokenAddress Address of the token.
+    function getPriceInUsd(address tokenAddress) public view returns (uint256) {
+        return (getAvaxPrice().mul(getPriceInAvax(tokenAddress))) / 1e18; // 18
+    }
+
+    /// @notice Get the price of a token in Avax.
+    /// @dev Need to be aware of decimals here, not always 18, it depends on the token.
+    /// @param tokenAddress Address of the token.
+    function getPriceInAvax(address tokenAddress)
+        public
+        view
+        returns (uint256)
+    {
+        if (tokenAddress == wavax) {
+            return 1e18;
+        }
+
+        IJoePair pair = IJoePair(joeFactory.getPair(tokenAddress, wavax));
+
+        (uint256 reserve0, uint256 reserve1, ) = pair.getReserves();
+        address token0Address = pair.token0();
+        address token1Address = pair.token1();
+
+        if (token0Address == wavax) {
+            reserve1 = reserve1.mul(_tokenDecimalsMultiplier(token1Address)); // 18
+            return (reserve0.mul(1e18)) / reserve1; // 18
+        } else {
+            reserve0 = reserve0.mul(_tokenDecimalsMultiplier(token0Address)); // 18
+            return (reserve1.mul(1e18)) / reserve0; // 18
+        }
+    }
+
+    /// @notice Calculates the multiplier needed to scale a token's numerical field to 18 decimals.
+    /// @param tokenAddress Address of the token.
+    function _tokenDecimalsMultiplier(address tokenAddress)
+        private
+        pure
+        returns (uint256)
+    {
+        uint256 decimalsNeeded = 18 - IJoeERC20(tokenAddress).decimals();
+        return 10**decimalsNeeded;
```

Comment:
could there ever be a case where `IJoeERC20(tokenAddress).decimals() > 18`?

Reply:
Theorically speaking, yes, but that would be dumb and I don't think we would add it to a farm.
18 decimals is already a lot x)